pkg/repstream/streampb/stream.proto

```diff
@@ -67,6 +68,8 @@ message StreamPartitionSpec {
 
     // Controls batch size in bytes.
     int64 batch_byte_size = 3;
+
+    roachpb.TenantID span_config_for_tenant = 4 [(gogoproto.nullable) = false];
```

Comment:
[followup-consideration] It could make sense to make a new protobuf message for use in the span stream.